Lib/typing.py

```diff
@@ -350,7 +350,11 @@ def _eval_type(t, globalns, localns, recursive_guard=frozenset()):
                 ForwardRef(arg) if isinstance(arg, str) else arg
                 for arg in t.__args__
             )
-            t = t.__origin__[args]
+            if (t.__origin__ is collections.abc.Callable
```

Comment:
I know this code pattern appears once in Generic code and once here now. Should we just convert this into one function? Maybe `_should_flatten_callable_args()`? Seems abit tiresome to repeat this (and it's also confusing for people who haven't dealt with Callable's args before).

Reply:
(sorry, missed this comment)
I added a helper function for this